File: src/Sample.VBasic/MinimalZeroInstall.vb
Imports ZeroInstall.Model
Imports ZeroInstall.Services

Module MinimalZeroInstall
    Sub Main(ByVal args As String())
        Run(New Requirements() With {.InterfaceID = args(0)})
    End Sub

    Private Sub Run(requirements As Requirements)
        Dim services = New ServiceLocator(New CliHandler())
        With services
            Dim selections = .Solver.Solve(requirements)
            Dim missing = .SelectionsManager.GetUncachedImplementations(selections)
            .Fetcher.Fetch(missing)
            .Executor.Start(selections)
        End With
    End Sub
End Module

File: src/Sample.IPython/MinimalZeroInstall.py
#import clr
#clr.AddReferenceToFile("ZeroInstall.Services.dll", "ZeroInstall.Model.dll")

import sys
from ZeroInstall import ServiceLocator, CliHandler
from ZeroInstall.Model import Requirements
from ZeroInstall.Injector import Executor

requirements = Requirements(InterfaceID = sys.argv[1]) # sys.argv[0]

services = ServiceLocator(CliHandler())
selections = services.Solver.Solve(requirements)
missing = services.SelectionsManager.GetUncachedImplementations(selections)
services.Fetcher.Fetch(missing)
services.Executor.Start(selections)
